extensions/vertx-graphql/deployment/src/main/java/io/quarkus/vertx/graphql/deployment/VertxGraphqlProcessor.java

```diff
@@ -18,4 +34,38 @@ FeatureBuildItem feature() {
     WebsocketSubProtocolsBuildItem websocketSubProtocols(CombinedIndexBuildItem combinedIndexBuildItem) {
         return new WebsocketSubProtocolsBuildItem("graphql-ws");
     }
+
+    @BuildStep
+    List<ReflectiveClassBuildItem> registerForReflection() {
+        return Arrays.asList(
+                new ReflectiveClassBuildItem(true, true,
+                        io.vertx.ext.web.handler.graphql.impl.GraphQLInputDeserializer.class.getName()),
+                new ReflectiveClassBuildItem(true, true, GraphQLBatch.class.getName()),
+                new ReflectiveClassBuildItem(true, true, GraphQLQuery.class.getName()));
+    }
+
+    @BuildStep
+    List<NativeImageResourceBuildItem> registerNativeImageResources() throws IOException {
+        ResolvedArtifact artifact = getVertxGraphQlArtifact();
+
+        List<NativeImageResourceBuildItem> nativeImageResourceBuildItems = new ArrayList<>();
+
+        File artifactFile = artifact.getArtifactPath().toFile();
+        try (JarFile jarFile = new JarFile(artifactFile)) {
+            Enumeration<JarEntry> entries = jarFile.entries();
+            while (entries.hasMoreElements()) {
+                JarEntry entry = entries.nextElement();
+                if (!entry.isDirectory() && entry.getName().startsWith("io/vertx/") && !entry.getName().endsWith(".class")) {
+                    nativeImageResourceBuildItems.add(new NativeImageResourceBuildItem(entry.getName()));
+                }
+            }
+        }
+
+        return nativeImageResourceBuildItems;
+    }
+
+    private ResolvedArtifact getVertxGraphQlArtifact() {
```

Comment:
I'm not entirely familiar with it but I wonder if we could use `applicationArchivesBuildItem`? It has a way to find the archive hosting a class and maybe we can go from there?

Not entirely sure though so I might be entirely wrong.

Reply:
Sure, I'll check it.

Reply:
Could not make it work with `ApplicationArchiveBuildItem`: I cannot find a `vertx graphql` archive.

Reply:
You need to use the AdditionalApplicationArchiveMarkerBuildItem to register a class/package in the archive as being a 'marker', and then this archive will be indexed. 

That said this approach also works fine, and avoids the need to index the archive.

Reply:
> You need to use the AdditionalApplicationArchiveMarkerBuildItem to register a class/package in the archive as being a 'marker', and then this archive will be indexed.
> 
> That said this approach also works fine, and avoids the need to index the archive.

Thanks @stuartwdouglas so I suppose we can keep it this way?

I'd try your suggestion for future references :-)

Reply:
After thinking about it I think a better approach would be to use something like:

getClass().getClassLoader().getResource("io/vertx/ext/web/handler/graphiql")

And then use the resulting URL to register all resources in the sub packages.

This could be generalised into a build item to make it easy to register all resources in a package/subpackage.

Reply:
Great, it joins Guillaume's suggestion https://github.com/quarkusio/quarkus/pull/5253#discussion_r343100496. I have pushed the changes. Let me know what you think.